source/jormungandr/jormungandr/schedule.py

```diff
@@ -169,7 +239,31 @@ def next_arrivals(self, request):
         return self.__stop_times(request, api=type_pb2.NEXT_ARRIVALS, arrival_filter=request["filter"])
 
     def next_departures(self, request):
-        return self.__stop_times(request, api=type_pb2.NEXT_DEPARTURES, departure_filter=request["filter"])
+        log = logging.getLogger(__name__)
+
+        resp = self.__stop_times(request, api=type_pb2.NEXT_DEPARTURES, departure_filter=request["filter"])
+        if request['data_freshness'] != 'realtime':
+            return resp
+
+        route_points = set()
+        for passage in resp.next_departures:
+            route_points.add(RoutePoint(stop_point=passage.stop_point, route=passage.route))
```

Comment:
I think it would be clearer with

````
route_points = {RoutePoint(stop_point=passage.stop_point, route=passage.route) for passage in resp.next_departures}
```